src/components/TrackFilePicker.test.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { render, fireEvent, waitFor, cleanup }  from '@testing-library/react';
+import {TrackFilePicker} from './TrackFilePicker';
+
+
+
+describe('TrackFilePicker', () => {
+    const testTracks = [{"files": [{"name": "fileA1.vg", "type": "graph"},
+                                   {"name": "fileA2.gbwt", "type": "haplotype"}]},
+                        {"files": [{"name": "fileB1.gbwt", "type": "haplotype"},
+                                   {"name": "fileB2.gam", "type": "read"}]},
+                        {"files": [{"name": "fileC1.xg", "type": "graph"}]}];
+
+    it('should render withouterrors', async () => {
+        const fakeOnChange = jest.fn();
+        const { getByText } = render(
+            <TrackFilePicker 
+            tracks={testTracks}
+            fileType={"graph"}
+            pickerType={"dropdown"}
+            handleInputChange={fakeOnChange}/>
+        );
+
+        // maybe need to search by "Select a file"
+        const placeholder = getByText("Select a file");
+        expect(placeholder).toBeTruthy();
+    });
+
+    it('should call onChange when an option is selected', async () => {
+        const fakeOnChange = jest.fn();
+        const { getByText, queryByTestId } = render(
+            <TrackFilePicker 
+            tracks={testTracks}
+            fileType={"haplotype"}
+            pickerType={"dropdown"}
+            handleInputChange={fakeOnChange}/>
+        );
+
+        // find div containing the component
+        const fileSelectComponent = queryByTestId('file-select-component');
+
+        expect(fileSelectComponent).toBeDefined();
+        expect(fileSelectComponent).not.toBeNull();
+
+        // expand the select box
+        fireEvent.keyDown(fileSelectComponent.firstChild, {key: "ArrowDown"});
+        await waitFor(() => getByText("fileB1.gbwt"));
+        fireEvent.click(getByText("fileB1.gbwt"));
+
+        expect(fakeOnChange).toHaveBeenCalledTimes(1);
+        expect(fakeOnChange).toHaveBeenCalledWith({"name": "fileB1.gbwt", "type": "haplotype"});
+
+        // make sure we can repeat the process
+        fireEvent.keyDown(fileSelectComponent.firstChild, {key: "ArrowDown"});
+        await waitFor(() => getByText("fileA2.gbwt"));
+        fireEvent.click(getByText("fileA2.gbwt"));
+
+        expect(fakeOnChange).toHaveBeenCalledTimes(2);
+        expect(fakeOnChange).toHaveBeenCalledWith({"name": "fileA2.gbwt", "type": "haplotype"});
+
+    });
+
+    it('should call onChange when queried by input value' ,async() => {
```

Comment:
I don't think we are querying by input value here. We are querying just for the `input` I think by tag name using a selector, and then we are *firing an event* to set the value.

```suggestion
    it('should call onChange when the input value is changed' ,async() => {
```